hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/metrics/BlobPublisher.java

```diff
@@ -39,10 +39,14 @@
 import static java.lang.Math.multiplyExact;
 
 /**
- * Renderer to serialize metrics to byte[] to be read by Management Center.
- * Additionally, it converts legacy metric names to {@code [metric=<oldName>]}.
+ * Implementation of {@link MetricsPublisher} which compresses metrics
+ * into a byte[] blob to be read by various consumers. The consumer itself
+ * is passed in as a parameter.
+ * <p>
+ * Before compressing it also converts legacy metric names to
+ * {@code [metric=<oldName>]}.
  */
-public class ManagementCenterPublisher implements MetricsPublisher {
+public class BlobPublisher implements MetricsPublisher {
```

Comment:
The BlobPublisher is a publisher itself and also a utility used inside InternalJobMetricsPublisher. It's kind of confusing. Would be better to extract a utility to do the compression and use it inside of ManCenterPublisher and JobMetricsPublisher (edited)

Reply:
👍